File: 0349_intersection_of_two_arrays.py
# https://leetcode-cn.com/problems/intersection-of-two-arrays/

class Solution:
    def intersection(self, nums1, nums2):
        """
        Inputs:
            nums1: list[int]
            nums2: list[int]
        Outputs:
            res: list[int]
        """

        # Check elements in nums1.
        tmp = {}
        for i in nums1:
            if not tmp.get(i):
                tmp[i] = 1

        # Check if these elements are in nums2.
        res = []
        for i in nums2:
            if tmp.get(i):
                res.append(i)
                tmp[i] = 0

        return res

File: Cpp/0349_intersection_of_two_arrays.cpp
// https://leetcode.cn/problems/intersection-of-two-arrays/

class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> check1, check2;
        vector<int> res;
        for (int num: nums1) {
            check1.insert(num);
        }
        for (int num: nums2) {
            check2.insert(num);
        }
        for (int num: check1) {
            if (check2.find(num) != check2.end()) {
                res.push_back(num);
            }
        }
        return res;
    }
};
